### ISSUE TITLE
Make toggle switch behave consistently with other UI toolkits

### ISSUE THREAD

**Issue by AndreasGaufer** [damonlynch/rapid-photo-downloader#45]
Launching rapid-photo-downloader but the "This Computer" Toggle is locked and can not be turned on. 

![grafik](https://user-images.githubusercontent.com/7531412/146547793-6b6b3b16-dca1-4424-a898-2a32a2974d25.png)

terminal shows
```

user@silentandy:~$ rapid-photo-downloader 
ERROR    Cannot set download destination display:  does not exist

(python3:48482): GVFS-RemoteVolumeMonitor-WARNING **: 13:55:32.533: remote volume monitor with dbus name org.gtk.vfs.UDisks2VolumeMonitor is not supported
^[[A^[[A^CTraceback (most recent call last):
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/offload.py", line 95, in <module>
Traceback (most recent call last):
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnaildaemon.py", line 234, in <module>
    offload.run()
Traceback (most recent call last):
Traceback (most recent call last):
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbloadbalancer.py", line 52, in <module>
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/offload.py", line 55, in run
    generate_thumbnails.run()
Traceback (most recent call last):
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 890, in <module>
    directive, content = self.receiver.recv_multipart()
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/zmq/sugar/socket.py", line 475, in recv_multipart
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 890, in <module>
    loadbalancer = ThumbnailLoadBalancer()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbloadbalancer.py", line 48, in __init__
    super().__init__("Thumbnail", ThumbnailLoadBalancerWorkerManager)
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/interprocess.py", line 651, in __init__
    parts = [self.recv(flags, copy=copy, track=track)]
  File "zmq/backend/cython/socket.pyx", line 791, in zmq.backend.cython.socket.Socket.recv
    thumbnail_extractor = ThumbnailExtractor()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 191, in __init__
  File "zmq/backend/cython/socket.pyx", line 827, in zmq.backend.cython.socket.Socket.recv
    thumbnail_extractor = ThumbnailExtractor()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 191, in __init__
    ioloop.IOLoop.instance().start()
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
  File "/usr/lib/python3/dist-packages/tornado/platform/asyncio.py", line 132, in start
    super().__init__("Thumbnail Extractor")
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/interprocess.py", line 1178, in __init__
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc
    self.asyncio_loop.run_forever()
  File "/usr/lib/python3.8/asyncio/base_events.py", line 570, in run_forever
KeyboardInterrupt
    super().__init__("Thumbnail Extractor")
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/interprocess.py", line 1178, in __init__
    self._run_once()
  File "/usr/lib/python3.8/asyncio/base_events.py", line 1823, in _run_once
    self.do_work()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 878, in do_work
    self.do_work()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 878, in do_work
    self.process_files()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 656, in process_files
    self.process_files()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 656, in process_files
    event_list = self._selector.select(timeout)
  File "/usr/lib/python3.8/selectors.py", line 468, in select
    directive, content = self.requester.recv_multipart()
  File "/usr/lib/python3/dist-packages/zmq/sugar/socket.py", line 475, in recv_multipart
    fd_event_list = self._selector.poll(timeout, max_ev)
Traceback (most recent call last):
KeyboardInterrupt
    directive, content = self.requester.recv_multipart()
  File "/usr/lib/python3/dist-packages/zmq/sugar/socket.py", line 475, in recv_multipart
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 890, in <module>
    parts = [self.recv(flags, copy=copy, track=track)]
  File "zmq/backend/cython/socket.pyx", line 791, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 827, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
    parts = [self.recv(flags, copy=copy, track=track)]
  File "zmq/backend/cython/socket.pyx", line 791, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc
Traceback (most recent call last):
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 890, in <module>
KeyboardInterrupt
    thumbnail_extractor = ThumbnailExtractor()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 191, in __init__
    super().__init__("Thumbnail Extractor")
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/interprocess.py", line 1178, in __init__
  File "zmq/backend/cython/socket.pyx", line 827, in zmq.backend.cython.socket.Socket.recv
    thumbnail_extractor = ThumbnailExtractor()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 191, in __init__
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc
    super().__init__("Thumbnail Extractor")
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/interprocess.py", line 1178, in __init__
    self.do_work()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 878, in do_work
KeyboardInterrupt
    self.do_work()
    self.process_files()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 878, in do_work
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 656, in process_files
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 890, in <module>
    directive, content = self.requester.recv_multipart()
  File "/usr/lib/python3/dist-packages/zmq/sugar/socket.py", line 475, in recv_multipart
    self.process_files()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 656, in process_files
    thumbnail_extractor = ThumbnailExtractor()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 191, in __init__
    parts = [self.recv(flags, copy=copy, track=track)]
  File "zmq/backend/cython/socket.pyx", line 791, in zmq.backend.cython.socket.Socket.recv
    directive, content = self.requester.recv_multipart()
  File "/usr/lib/python3/dist-packages/zmq/sugar/socket.py", line 475, in recv_multipart
    super().__init__("Thumbnail Extractor")
  File "zmq/backend/cython/socket.pyx", line 827, in zmq.backend.cython.socket.Socket.recv
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/interprocess.py", line 1178, in __init__
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnaildaemon.py", line 106, in run
Traceback (most recent call last):
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 890, in <module>
    parts = [self.recv(flags, copy=copy, track=track)]
  File "zmq/backend/cython/socket.pyx", line 791, in zmq.backend.cython.socket.Socket.recv
    directive, content = self.receiver.recv_multipart()
  File "/usr/lib/python3/dist-packages/zmq/sugar/socket.py", line 475, in recv_multipart
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
  File "zmq/backend/cython/socket.pyx", line 827, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc
    thumbnail_extractor = ThumbnailExtractor()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 191, in __init__
    parts = [self.recv(flags, copy=copy, track=track)]
KeyboardInterrupt
  File "zmq/backend/cython/socket.pyx", line 791, in zmq.backend.cython.socket.Socket.recv
KeyboardInterrupt
    super().__init__("Thumbnail Extractor")
  File "zmq/backend/cython/socket.pyx", line 827, in zmq.backend.cython.socket.Socket.recv
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/interprocess.py", line 1178, in __init__
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
    self.do_work()
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 878, in do_work
Traceback (most recent call last):
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/renameandmovefile.py", line 1099, in <module>
    self.do_work()
KeyboardInterrupt
    self.process_files()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 656, in process_files
    rename.run()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/renameandmovefile.py", line 999, in run
    directive, content = self.requester.recv_multipart()
  File "/usr/lib/python3/dist-packages/zmq/sugar/socket.py", line 475, in recv_multipart
    directive, content = self.receiver.recv_multipart()
  File "/usr/lib/python3/dist-packages/zmq/sugar/socket.py", line 475, in recv_multipart
    parts = [self.recv(flags, copy=copy, track=track)]
  File "zmq/backend/cython/socket.pyx", line 791, in zmq.backend.cython.socket.Socket.recv
    parts = [self.recv(flags, copy=copy, track=track)]
  File "zmq/backend/cython/socket.pyx", line 791, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 827, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 827, in zmq.backend.cython.socket.Socket.recv
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 878, in do_work
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc
KeyboardInterrupt
KeyboardInterrupt
    self.process_files()
  File "/home/rootgauf/.local/lib/python3.8/site-packages/raphodo/thumbnailextractor.py", line 656, in process_files
    directive, content = self.requester.recv_multipart()
  File "/usr/lib/python3/dist-packages/zmq/sugar/socket.py", line 475, in recv_multipart
    parts = [self.recv(flags, copy=copy, track=track)]
  File "zmq/backend/cython/socket.pyx", line 791, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 827, in zmq.backend.cython.socket.Socket.recv
  File "zmq/backend/cython/socket.pyx", line 186, in zmq.backend.cython.socket._recv_copy
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc
```
i hestitate to upload the debug tar here beacause it has a lot of personal filenames in the logs... happy to email them privately.

**Comment by damonlynch** [comment]
Yes, you need to email me the tar file because the only thing the traceback above tells me is that you hit Ctrl-C to terminate the program. Hopefully the log file contains the actual problem cause. Thanks.

**Comment by damonlynch** [comment]
Closing due to lack of response. Will reopen if I receive the files necessary to diagnose the problem.

**Comment by damonlynch** [comment]
Unfortunately there is nothing in the log files to show what went wrong. It seems the splash screen did not close before the problem appeared, yes? If you tried it again and the problem came back, have you tried resetting the preferences like this, and trying again?:

```rapid-photo-downloader --reset```

(Before resetting the prefes you may like to save your custom file renaming & subfolder generation presets, found in the file `~/.config/Rapid Photo Downloader/Rapid Photo Downloader.conf`)

**Comment by AndreasGaufer** [comment]
Ok, this issue is **solved**. 

Basically, the user has to **grab the knob and drag it over** to the desired side **or click into the space next to the knob** to move it. 

Simply clicking the Knob has no effect and shows now UI change at all, which may make it appear unresponsive or dysfunctional.

**Comment by damonlynch** [comment]
Thank you for that excellent feedback. I just tried a toggle switch in Gnome and indeed, when you click on the button itself, the switch moves. No need to click in the space in the control to the left or right of the button.

**Comment by AndreasGaufer** [comment]
I confirm this fixed in 0.9.28 in my environment. Thank You so much!
